test/src/graph/MaskRaysTest.cpp

```diff
@@ -0,0 +1,98 @@
+
+	void initializeMask(int raysCount)
+	{
+		points_mask.resize(raysCount);
+		std::fill(points_mask.begin(), points_mask.end(), 1);
+		for (int i = 0; i < maskCount; i++) {
+			points_mask[i] = 0;
+		}
+	}
+};
+
+TEST_F(MaskRaysTest, invalid_argument_node) { EXPECT_RGL_INVALID_ARGUMENT(rgl_node_raytrace_configure_mask(nullptr, nullptr, 0)); }
+
+TEST_F(MaskRaysTest, invalid_argument_mask)
+{
+	rgl_node_t raytraceNode = nullptr;
+	EXPECT_RGL_SUCCESS(rgl_node_raytrace(&raytraceNode, nullptr));
+
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_node_raytrace_configure_mask(raytraceNode, nullptr, 0));
+}
+
+TEST_F(MaskRaysTest, invalid_argument_count)
+{
+	rgl_node_t raytraceNode = nullptr;
+	EXPECT_RGL_SUCCESS(rgl_node_raytrace(&raytraceNode, nullptr));
+
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_node_raytrace_configure_mask(raytraceNode, points_mask.data(), 0));
+}
+
+TEST_F(MaskRaysTest, valid_arguments)
+{
+	const int raysNumber = 100;
+	rgl_node_t raytraceNode = nullptr;
+	EXPECT_RGL_SUCCESS(rgl_node_raytrace(&raytraceNode, nullptr));
+
+	// Mask
+	initializeMask(raysNumber);
+
+	EXPECT_RGL_SUCCESS(rgl_node_raytrace_configure_mask(raytraceNode, points_mask.data(), raysNumber));
+}
+
+TEST_F(MaskRaysTest, use_case)
+{
+	// Scene
+	rgl_entity_t wall = makeEntity(makeCubeMesh());
+	rgl_mat3x4f wallPose = Mat3x4f::TRS(Vec3f(0,0,0), Vec3f(0, 0, 0), Vec3f(10,10,10)).toRGL();
+	EXPECT_RGL_SUCCESS(rgl_entity_set_pose(wall, &wallPose));
+
+	// Rays
+	std::vector<rgl_mat3x4f> rays = makeLidar3dRays(360, 180, 0.36, 0.18);
+
+	const int raysNumber = rays.size();
+	// Mask
+	initializeMask(raysNumber);
+
+	// Graph
+	rgl_node_t useRaysNode = nullptr;
+	rgl_node_t raytraceNode = nullptr;
+	rgl_node_t compactNode = nullptr;
+	rgl_node_t yieldNode = nullptr;
+
+	// Prepare graph without filtering
+	EXPECT_RGL_SUCCESS(rgl_node_rays_from_mat3x4f(&useRaysNode, rays.data(), rays.size()));
+	EXPECT_RGL_SUCCESS(rgl_node_raytrace(&raytraceNode, nullptr));
+	EXPECT_RGL_SUCCESS(rgl_node_points_compact_by_field(&compactNode, IS_HIT_I32));
+	EXPECT_RGL_SUCCESS(rgl_node_points_yield(&yieldNode, fields.data(), fields.size()));
+
+	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(useRaysNode, raytraceNode));
+	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(raytraceNode, compactNode));
+	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(compactNode, yieldNode));
+
+	EXPECT_RGL_SUCCESS(rgl_graph_run(raytraceNode));
+
+	TestPointCloud outputPointCloud = TestPointCloud::createFromNode(yieldNode, fields);
+	auto fullCloudSize = outputPointCloud.getPointCount();
+
+	EXPECT_RGL_SUCCESS(rgl_node_raytrace_configure_mask(raytraceNode, points_mask.data(), raysNumber));
+
+	EXPECT_RGL_SUCCESS(rgl_graph_run(raytraceNode));
+	TestPointCloud outputPointCloudMasked = TestPointCloud::createFromNode(yieldNode, fields);
+	auto maskedCloudSize = outputPointCloudMasked.getPointCount();
+
+	EXPECT_EQ(fullCloudSize - maskCount, maskedCloudSize);
```

Comment:
You can also easily verify if the proper rays are hidden using RAY_IDX field.

Reply:
I can.
Not rly sure if I want in this test.